core/storage/blog_post/gae_models.py

```diff
@@ -0,0 +1,457 @@
+    url_fragment = (
+        datastore_services.StringProperty(indexed=True, required=True))
+    # Tags associated with the blog post.
+    tags = datastore_services.StringProperty(indexed=True, repeated=True)
+    # The thumbnail filename of the blog post.
+    thumbnail_filename = datastore_services.StringProperty(required=True)
+    # Time when the blog post model was published.
+    published_on = (
+        datastore_services.DateTimeProperty(indexed=True))
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model contains data to pseudonymize corresponding to a user:
+        author_id field.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether BlogPostSummaryModel exists for user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            cls.author_id == user_id
+        ).get(keys_only=True) is not None
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model data has already been associated as a part of the
+        BlogPostModel to the user and thus does not need a separate user
+        association.
+        """
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model contains data corresponding to a user (author_id), but this
+        isn't exported because noteworthy details that belong to this
+        model have already been exported as a part of the BlogPostModel.
+        """
+        return dict(super(BlogPostSummaryModel, cls).get_export_policy(), **{
+            'author_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'title': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'summary': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'url_fragment': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'tags': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'thumbnail_filename': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'published_on': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+        })
+
+    @classmethod
+    def create(cls, blog_post_id, author_id):
+        """Creates a new BlogPostSummaryModel entry.
+
+        Args:
+            blog_post_id: str. Blog Post ID of the newly-created blog post.
+            author_id: str. User ID of the author.
+
+        Returns:
+            BlogPostSummaryModel. The newly created BlogPostSummaryModel
+            instance.
+
+        Raises:
+            Exception. A blog post summary model with the given blog post ID
+                exists already.
+        """
+        if cls.get_by_id(blog_post_id):
+            raise Exception(
+                'Blog Post ID conflict on creating new blog post summary model.'
```

Comment:
I don't think you should have a create() here, actually. The summary model should be autogenerated, right? Which means that, at the point it's autogenerated, you'll already have all the data for it, so you don't actually need a create method that populates it with dummy values to begin with.

Reply:
The summary of the blog post will be autogenerated. However, if the user creates a new blog but closes the window without saving it as a draft or publishing it, We will have a blog post model created but it will have no summary model. Then while generating blog cards on the blog dashboard, it may go untracked, as we will query all the blog post summary models using the user id to display the cards. So I thought it to be better to generate the summary model at the same time and update it accordingly.

Reply:
Oh I see. But then can you generate the summary model the usual way when you create the blog post model (by using properties from the blog post model in the same way that you would in general), rather than expose a general special-case create_summary() function? That would be simpler and prevent others from calling this erroneously.

Reply:
Done. Removed the create method. WIll autogenerate the model when the blog post model is generated.